consensus/core/src/validator.rs

```diff
@@ -1,36 +1,49 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::block_verifier::BlockVerifier;
-use crate::metrics::{initialise_metrics, Metrics};
-use consensus_config::{AuthorityIndex, Committee, Parameters, ProtocolKeyPair};
-use prometheus::Registry;
 use std::sync::Arc;
 use std::time::Instant;
+
+use consensus_config::{AuthorityIndex, Committee, Parameters, ProtocolKeyPair};
+use prometheus::Registry;
+use sui_protocol_config::ProtocolConfig;
 use tracing::info;
 
+use crate::block_verifier::BlockVerifier;
+use crate::context::Context;
+use crate::metrics::initialise_metrics;
+
 pub struct Validator {
+    context: Arc<Context>,
     start_time: Instant,
-    metrics: Arc<Metrics>,
 }
 
 impl Validator {
```

Comment:
Should we use validator or authority more consistently across the codebase, to refer to a consensus instance? Picking either is fine by me.

Reply:
hhm good point. Using `Authority` makes sense and in the same time it can be confusing as well as it gets quite overloaded. What about `Consensus`? Is it too ambiguous?

Reply:
I think we should use `Authority` or `ConsensusAuthority` to differentiate from sui auhtority. Then in the codebase we can just refer to authority

Reply:
Since the type will be in `consensus-core` crate, maybe we will be able to avoid confusion with Sui authority, similar to Sui vs Narwhal committee? I will open another PR for discussion and change.

Reply:
Good idea, let's have this discussion there